inc/StyleUtils.php

```diff
@@ -100,6 +92,152 @@ public function cssStyleini($tpl, $preview=false) {
+            if(file_exists($ini)) {
+                $data = parse_ini_file($ini, true);
+                // FIXME does preview mode affect replacements only, or stylesheets too?
+                // replacements
+                if(is_array($data['replacements'])) {
+                    $mergedinis['replacements'] = array_merge(
+                        $mergedinis['replacements'],
+                        $this->cssFixreplacementurls($data['replacements'], $webbase)
+                    );
+                }
+            }
+        }
+
+        return array(
+            'stylesheets' => $mergedinis['stylesheets'],
+            'replacements' => $mergedinis['replacements']
+        );
+    }
+
+    protected function getStyleIni() {
+        static $inis = null;
+        if ( !$inis ) {
+            $inis = $this->retrieveStyles('styleini','parse_ini_file', array(true));
+
+            // FIXME is this trim filter still necessary?
+            foreach ($inis as $key => &$section) {
+                if ($section !== 'replacements') {
+                    continue;
+                }
+                $section = array_map('trim', $section);
+                $section = preg_replace('/^#.*/', '', $section);
+                $section = array_filter($section);
+            }
+        }
+        return $inis;
+    }
+
+    protected function retrieveStyles($type, $fn, $params = array()) {
+        global $conf;
+        global $config_cascade;
+        $incbase = tpl_incdir();
+        $stylesheets = array(); // mode, file => base
+
+        // guaranteed placeholder => value
+        $replacements = array(
+            '__text__' => "#000",
+            '__background__' => "#fff",
+            '__text_alt__' => "#999",
+            '__background_alt__' => "#eee",
+            '__text_neu__' => "#666",
+            '__background_neu__' => "#ddd",
+            '__border__' => "#ccc",
+            '__highlight__' => "#ff9",
+            '__link__' => "#00f",
+        );
+
+        $combined = [];
+
+        if (!is_array($config_cascade[$type])) {
+            trigger_error('Missing config cascade for "'.$type.'"',E_USER_WARNING);
+        }
+
+        foreach (array('default','local','protected') as $config_group) {
+
+            if (empty($config_cascade[$type][$config_group])) continue;
+
+            foreach ($config_cascade[$type][$config_group] as $file) {
+                // replace the placeholder with the name of the current template
+                $file = str_replace('TPL_PLACEHOLDER', $conf['template'], $file);
+
+                if (file_exists($file)) {
+                    $config = call_user_func_array($fn, array_merge(array($file), $params));
+
+                    if (is_array($config['stylesheets'])) {
+                        foreach($config['stylesheets'] as $file => $mode) {
+
+                            // set proper server dirs
+                            $webbase = $this->getServerPath($config_group);
```

Comment:
The `$webbase` should be set outside the two conditionals, otherwise it could happen that it's not available in the second one.